synapse/dendrite/coordinator.py

```diff
@@ -0,0 +1,261 @@
+        for queue in self.queues(mimeType):
+            logger.debug('Adding job:put in queue %s', queue)
+            self.jobs.put(queue, job)
+
+    def _getMimeType(self, bytes):
+        mimeType = None
+        with magic.Magic(flags=magic.MAGIC_MIME_TYPE) as mimer:
+            try:
+                mimeType = mimer.id_buffer(bytes)
+            except magic.MagicError as e:
+                logger.error('Error determining mime type %s', e)
+        return mimeType
+
+    def _assertOpts(self):
+        if not Coordinator._REQUIRED_ARGS.issubset(self.opts):
+            raise TypeError("kwargs missing required args: %s" % (Coordinator._REQUIRED_ARGS - set(self.opts.keys())))
+
+    def _assertConnected(self):
+        try:
+            self.listenCore.getTufoByIden('')
+            logger.info('Successfully connected to cortex')
+        except s_exc.NoSuchObj:
+            raise s_exc.NoSuchObj(msg='unable to connect to cortex, please check url')
+        try:
+            self.axon.alloc(0)
+            logger.info('Successfully connected to axon')
+        except s_exc.NoSuchObj:
+            raise s_exc.NoSuchObj(msg='unable to connect to axon, please check url')
+        try:
+            self.jobs.isEmpty('')
+            logger.info('Successfully connected to job queue')
+        except s_exc.NoSuchObj:
+            raise s_exc.NoSuchObj(msg='unable to connect to job queue, please check url')
+
+    def _initCortex(self):
+        coreurl = self.opts.get('interncore', 'ram:///')
+        logger.debug('Managing parser registrations via internal cortex with url %s', coreurl)
+        self.core = s_cortex.openurl(coreurl)
+        self.core.addTufoForm('dendrite:coordinator', ptype='guid')
+        self.core.addTufoProp('dendrite:coordinator', 'mime', ptype='str', req=True)
+        self.core.addTufoProp('dendrite:coordinator', 'queue', ptype='str', req=True)
+
+        self.core.addTufoForm('dendrite:state', ptype='guid')
+        self.core.addTufoProp('dendrite:state', 'mintime', ptype='int')
+
+    def _initState(self):
+        self.state = self.core.getTufoByProp('dendrite:state', self._stateGuid())
+        mintime = self.opts.get('mintime')
+        if not self.state:
+            if not mintime:
+                mintime = int(time.time() * 1000)
+            self.state = self.core.formTufoByProp('dendrite:state', self._stateGuid(), mintime=mintime)
+        elif mintime:
+            self.core.setTufoProps(self.state, mintime=mintime)
+
+    def _updateState(self):
+        # currently, the tstamp on a tufo is only set when a tufo is created, and not when a prop is updated. For now
+        # just record current time as a point of return on restart... there may exist tufos that had prop updates during
+        # down time, but at least all new tufos will be processed.
+        self.core.setTufoProps(self.state, mintime=int(time.time() * 1000))
+
+    def _stateGuid(self):
```

Comment:
Yeah... I paused during implementation not really knowing the synaptic approach here. I just wanted to use the internal cortex(which is intended to persist across runs) to keep track of some 'point in time' state to be able to pick-up where the Coordinator left off across runs. Please let me know if there is a better way to achieve this.